packages/client/src/components/AppNavBar/AppNavBar.tsx

```diff
@@ -5,199 +5,215 @@ import MenuIcon from '@mui/icons-material/Menu';
-          <IconButton size='large' onClick={handleMenuOpen} sx={{ p: 0 }}>
-            <MenuIcon color='primary' fontSize='large' />
-          </IconButton>
-          <Menu
-            anchorEl={menuAnchorElem}
-            anchorOrigin={{
-              vertical: 'bottom',
-              horizontal: 'left',
-            }}
-            open={Boolean(menuAnchorElem)}
-            onClose={handleMenuClosed}
-          >
-            {menuItems(auth.isLoggedIn ? LEFT_PAGES : PAGES)}
-          </Menu>
-        </Box>
-        <Link href={routes.home} sx={{ mx: 'auto' }}>
-          <img
-            src={theme.palette.mode === 'dark' ? logoDark : logo}
-            alt='JeszCoChcesz'
-            style={{ display: 'block', height: '6rem', paddingBlock: '1rem' }}
-          />
-        </Link>
-        <Box sx={{ flexGrow: 1, ml: 8, display: { xs: 'none', md: 'block' } }}>{LEFT_PAGES.map(pageToButton)}</Box>
-        {!auth.isLoggedIn && <Box sx={{ display: { xs: 'none', md: 'block' } }}>{RIGHT_PAGES.map(pageToButton)}</Box>}
-        <Tooltip title={`tryb ${theme.palette.mode === 'dark' ? 'jasny' : 'ciemny'}`} placement='top'>
-          <IconButton
-            sx={{ ml: 1, color: theme.palette.primary.main }}
-            onClick={colorMode.toggleColorMode}
-            color='inherit'
-          >
-            {theme.palette.mode === 'dark' ? <Brightness7Icon /> : <Brightness4Icon />}
-          </IconButton>
-        </Tooltip>
-        {auth.isLoggedIn && (
-          <Box sx={{ display: 'block' }}>
-            <IconButton onClick={handleProfileMenuOpen} sx={{ p: 2 }} title='Profil'>
-              <PersonIcon sx={{ fontSize: '3.5rem' }} color='secondary' />
-            </IconButton>
-            <Menu
-              anchorEl={profileMenuAnchorElem}
-              anchorOrigin={{
-                vertical: 'bottom',
-                horizontal: 'left',
-              }}
-              open={Boolean(profileMenuAnchorElem)}
-              onClose={handleProfileMenuClosed}
-            >
-              {[
-                auth.userRole === Role.User ? menuItems(USER_PAGES) : menuItems(PARTNER_PAGES),
-                <Divider key='divider' />,
-                <MenuItem key='logout' onClick={handleLogout}>
-                  Wyloguj się
-                </MenuItem>,
-              ]}
-            </Menu>
-          </Box>
-        )}
-        <Box sx={{ backgroundColor: theme.palette.secondary.main, borderRadius: '50%', ml: 2 }}>
-          <IconButton href={routes.shoppingCart} sx={{ p: 2 }} title='Koszyk'>
-            <ShoppingBasketIcon color='primary' />
-          </IconButton>
-        </Box>
-      </Toolbar>
-    </AppBar>
-  );
+	const auth = useAuth();
+	const colorMode = useContext(ThemeContext);
+	const theme = useTheme();
+	const { cart, sumAllCartDishes } = useShoppingCart();
+	const [dishesSum, setDishesSum] = useState<number>(0);
+
+	useEffect(() => {
+		setDishesSum(sumAllCartDishes(cart));
+	}, [cart, sumAllCartDishes]);
```

Comment:
```suggestion

  const dishesSum = sumAllCartDishes(cart);
```

Możesz takie wartości wyliczać zwyczajnie, używasz o 2 hooki mniej i potencjalnie unikasz dodatkowych rerenderów. Możesz dorzucić `useMemo` jeśli martwisz się o wydajność.

- https://prateeksurana.me/blog/why-you-should-avoid-using-state-for-computed-properties/
- https://www.robinwieruch.de/react-computed-properties/

---

Możesz też zwracać od razu `dishesSum` z kontekstu (jak chyba kiedyś próbowałaś), nawet będzie lepiej, bo obecnie zwracana jest funkcja, która jest pure i nigdy się nie zmienia (więc nie ma potrzeby przekazywać jej kontekstem).

Wtedy analogicznie, przerzucasz linijkę którą tu wrzuciłem do kontekstu i zmieniasz w `value` `sumAllCartDishes` na `dishesSum`.